src/panels/history/ha-panel-history.ts

```diff
@@ -391,19 +312,24 @@ class HaPanelHistory extends SubscribeMixin(LitElement) {
           }
         }
 
-        const foundDevices = this._areaIdToDevices[singleSearchingAreaId];
-        if (foundDevices) {
-          for (const foundDevice of foundDevices) {
-            const foundDeviceEntities =
-              this._deviceIdToEntities[foundDevice.id];
-            for (const foundDeviceEntity of foundDeviceEntities) {
-              if (
-                (!foundDeviceEntity.area_id ||
-                  foundDeviceEntity.area_id === singleSearchingAreaId) &&
-                foundDeviceEntity.entity_category === null
-              ) {
-                entityIds.add(foundDeviceEntity.entity_id);
-              }
+        const foundDevices = this._areaDeviceLookup[singleSearchingAreaId];
+        if (!foundDevices) {
```

Comment:
Most of the time you're checking for !x?.length, any reason to do this instead? Technically the length is not necessary anywhere, it should just remove the need to create an empty iterator at each for loop, but my OCD made me type this comment :)

Reply:
```suggestion
        if (!foundDevices.length) {
```

Reply:
I rarely use Github so I'm not familiar with suggesting changes, but this should be !foundDevices?.length